MBINCompiler/Models/Structs/GcRewardTableItem.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Runtime.InteropServices;
+
+namespace MBINCompiler.Models.Structs
+{
+    public class GcRewardTableItem : NMSTemplate
+    {
+        public float PercentageChance;
+        public object Reward; // Generic
```

Comment:
I'd replace **object** with **NMSTemplate** here.
We should also add a custom attribute to specify which concrete NMSTemplates are allowed here.